src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java

```diff
@@ -5,28 +5,67 @@
-
-    static final String KUSTO_SINK_TEMPDIR = "kusto.sink.tempdir";
-    static final String KUSTO_SINK_FLUSH_SIZE = "kusto.sink.flush_size";
-    static final String KUSTO_SINK_FLUSH_INTERVAL_MS = "kusto.sink.flush_interval_ms";
-    static final String KUSTO_SINK_WRITE_TO_FILES = "kusto.sink.write_to_files";
+    static final String KUSTO_AUTH_USERNAME_CONF = "kusto.auth.username";
+    private static final String KUSTO_AUTH_USERNAME_DOC = "Kusto username for authentication, also configure kusto.auth.password.";
+    private static final String KUSTO_AUTH_USERNAME_DISPLAY = "Kusto Auth Username";
+    
+    static final String KUSTO_AUTH_PASSWORD_CONF = "kusto.auth.password";
+    private static final String KUSTO_AUTH_PASSWORD_DOC = "Kusto password for the configured username.";
+    private static final String KUSTO_AUTH_PASSWORD_DISPLAY = "Kusto Auth Password";
+    
+    static final String KUSTO_AUTH_APPID_CONF = "kusto.auth.appid";
+    private static final String KUSTO_AUTH_APPID_DOC = "Configure this to perform authentication using Kusto AppID, "
+        + "also configure kusto.auth.appkey and kusto.auth.authority.";
+    private static final String KUSTO_AUTH_APPID_DISPLAY = "Kusto Auth AppID";
+    
+    static final String KUSTO_AUTH_APPKEY_CONF = "kusto.auth.appkey";
+    private static final String KUSTO_AUTH_APPKEY_DOC = "Kusto AppKey for authentication.";
+    private static final String KUSTO_AUTH_APPKEY_DISPLAY = "Kusto Auth AppKey";
+    
+    static final String KUSTO_AUTH_AUTHORITY_CONF = "kusto.auth.authority";
+    private static final String KUSTO_AUTH_AUTHORITY_DOC = "Kusto authority for authentication.";
+    private static final String KUSTO_AUTH_AUTHORITY_DISPLAY = "Kusto Auth Authority";
+    
+    static final String KUSTO_TABLES_MAPPING_CONF = "kusto.tables.topics.mapping";
+    private static final String KUSTO_TABLES_MAPPING_DOC = "Kusto target tables mapping (per topic mapping, "
+        + "'topic1:table1;topic2:table2;').";
+    private static final String KUSTO_TABLES_MAPPING_DISPLAY = "Kusto Table Topics Mapping";
+    
+    static final String KUSTO_SINK_TEMP_DIR_CONF = "kusto.sink.tempdir";
+    private static final String KUSTO_SINK_TEMP_DIR_DOC = "Temp dir that will be used by kusto sink to buffer records. "
+        + "defaults to system temp dir.";
+    private static final String KUSTO_SINK_TEMP_DIR_DISPLAY = "Temporary Directory";
+    
+    static final String KUSTO_SINK_FLUSH_SIZE_BYTES_CONF = "flush.size.bytes";
+    private static final String KUSTO_SINK_FLUSH_SIZE_BYTES_DOC = "Kusto sink max buffer size (per topic+partition combo).";
+    private static final String KUSTO_SINK_FLUSH_SIZE_BYTES_DISPLAY = "Maximum Flush Size";
+    
+    static final String KUSTO_SINK_FLUSH_INTERVAL_MS_CONF = "flush.interval.ms";
+    private static final String KUSTO_SINK_FLUSH_INTERVAL_MS_DOC = "Kusto sink max staleness in milliseconds (per topic+partition combo).";
```

Comment:
Isnt this size per topic+partition+task?

Reply:
Same as above.